### ISSUE TITLE
Tagging of sections does not play well with validation template

### ISSUE THREAD

**Issue by robertodr** [dev-cafe/parselglossy#46]
The following is parsed by the grammar, but does not pass validation.
```
name<foo> {
   ... keywords ...
  .... sections ...
}
```
 This totally slipped through the cracks and I am unsure how to solve it.

**Comment by bast** [comment]
I am not sure I see the problem yet. Do you have an explicit example that should pass validation but does not? (maybe the above is an explicit example)

**Comment by robertodr** [comment]
Yes, the above example fails with `name<foo>` section not recognized. The idea with tagging is too have multiple similar sections. They should all be validated in the same way and the tag should only have meaning for the final consumer. I think it will be enough to match section names with a regex, though how this will ripple through (if at all) I cannot foresee.

**Comment by robertodr** [comment]
I think we need to touch all the recusrive functions to be tag-tolerant, but there's a little nasty twist. 
For example, the `rec_merge_ours` function around here: https://github.com/dev-cafe/parselglossy/blob/master/parselglossy/validation_plumbing.py#L117
The `set` difference needs to be tolerant with respect to tagging (only strings enclosed by`<>` allowed), but only for sections, as tagged keyword make no sense, IMO.

I don't consider tagging to be an enormously important feature: it can be emulated by duplication in the validation template.

**Comment by bast** [comment]
The minimum would be a mention in docs and not support this.

We may consider implementing keyword/section "templates" which would help avoiding input template code repetition.

**Comment by robertodr** [comment]
Still low priority, as there is not really a relevant use case so far.
